### ISSUE TITLE
[4.x] Blueprint Modals Accessibility

### ISSUE THREAD

**Issue by AndreasSchantl** [statamic/cms#7786]
### Bug description

The new modals a pretty cool. But to make them more accessible, they should be focus trapped. 

### How to reproduce

Any new modal within the blueprint builder.

### Environment

```yaml
Environment
Application Name: Laravel
Laravel Version: 9.52.4
PHP Version: 8.1.17
Composer Version: 2.2.21
Environment: local
Debug Mode: ENABLED
URL: localhost
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Drivers
Broadcasting: log
Cache: statamic
Database: mysql
Logs: stack / single
Mail: smtp
Queue: sync
Session: file

Statamic
Addons: 0
Antlers: runtime
Stache Watcher: Enabled
Static Caching: Disabled
Version: 4.0.0-alpha.1 Solo
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

None
